docs/src/components/InlineFilter.tsx

```diff
@@ -1,21 +1,40 @@
-import { Fragment } from 'react';
-import { FilterChildren } from './FilterChildren';
+import { useRouter } from 'next/router';
+import { Framework } from '@/data/frameworks';
 
 type InlineFilterProps = {
   children: React.ReactNode;
-  filters: string[];
+  /**
+   * List of platforms that will render this content. If the current platform
+   * is not in this list then the content will not be render
+   */
+  filters: (Framework | 'all')[];
 };
 
+/**
+ *
+ * Usage:
+ * <InlineFilter filters={['react', 'android']}>
+ *  This content will only render for react and angular platforms
+ * </InlineFilter>
+ */
 export const InlineFilter = ({ filters, children }: InlineFilterProps) => {
   if (!filters || !Array.isArray(filters) || filters.length < 1) {
-    return <></>;
+    return null;
   }
+  const router = useRouter();
 
-  const filteredChildren: Array<React.JSX.Element> = [];
+  let filterKey = '';
 
-  filters.forEach((filter) => {
-    filteredChildren.push(<Fragment key={filter}>{children}</Fragment>);
-  });
+  if ('platform' in router.query) {
+    filterKey = router.query.platform as string;
+  }
+
+  for (let i = 0; i < filters.length; i++) {
```

Comment:
nit: could make use of for of here, so we don't have to track indexes:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of